openslides/core/static/js/core/site.js

```diff
@@ -752,6 +753,45 @@ angular.module('OpenSlidesApp.core.site', [
     };
 })
 
+// counter of new (unread) chat messages
+.value('NewChatMessages', [])
+
+// ChatMessage Controller
+.controller('ChatMessageCtrl', [
+    '$scope',
+    '$http',
+    'ChatMessage',
+    'NewChatMessages',
+    function ($scope, $http, ChatMessage, NewChatMessages) {
+        ChatMessage.bindAll({}, $scope, 'chatmessages');
+        $scope.unreadMessages = NewChatMessages.length;
+        $scope.chatboxIsCollapsed = true;
+        $scope.openChatbox = function () {
+            $scope.chatboxIsCollapsed = !$scope.chatboxIsCollapsed;
+            NewChatMessages = [];
+            $scope.unreadMessages = NewChatMessages.length;
+        }
+        $scope.sendMessage = function () {
+            $http.post(
+                '/rest/core/chatmessage/',
+                {message: $scope.newMessage}
+            )
+            .success(function () {
+                $scope.newMessage = '';
```

Comment:
If you have a poor connection to the server, then the message in the text disappears not until the request returned a success message. In this case the user could hit the send button multiple times and the message would be send to the server more then once. We need a solution for this. For example, the server should skip multipe messages with the same content or you should disable the send button until the request returns a success or error message